### ISSUE TITLE
Mask strategy 

### ISSUE THREAD

**Issue by lemon234071** [ChenRocks/Distill-BERT-Textgen#2]
I am confused with the masking strategy described in the supplementary, is following right?:

1, You repeat every training sample for 7 times.
2, Then manually mask the position as the figure show. so each sample of the 7 may mask about 1-2 tokens?  Then the mask token may approximates 15% in one sample of the 7?
3, When compute the logits for distillation, you may get 1-2 logit of words ? then you save these logits.

**Comment by ChenRocks** [comment]
Thanks for you interest in our project.

This masking strategy is used at the final distillation stage. For example, after we have a C-MLM fine-tuned on En-Vi translation, we will use these masking strategy on all the En-Vi data and then store only the masked position's logits. Please see Figure 4 in the appendix, every word in the target sentence (Vi) will be masked exactly once. Therefore, the 7-time repeated example will produce all logits for 1 En-Vi example and can be used for distillation. We store these logits as preprocessing step so that we won't need to run repeated inference at the final seq2seq distillation training.

I hope this explanation helps for now. Stay tuned for our code release in the coming weeks.

**Comment by ChenRocks** [comment]
If you are curious about C-MLM fine-tuning masking strategy, it's standard BERT masking on target language with 15% probability.

**Comment by lemon234071** [comment]
> If you are curious about C-MLM fine-tuning masking strategy, it's standard BERT masking on target language with 15% probability.

Thank you , your answer is very clear!

**Comment by ChenRocks** [comment]
Code has been released so closing this.
